### PR TITLE
feat: kubernetes deployment target

### PR DESCRIPTION
**Pull request by domenicsim1**

### CODE REVIEW

**Comment by benPearce1** [comment]
there is a common `deployment-target view` command that attempts to defer to the specific implementations, this will need to be updated for to use the k8s view

File: pkg/cmd/target/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdKubernetes(f factory.Factory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "kubernetes <command>",
+		Short:   "Manage Kubernetes deployment targets",
+		Long:    "Manage Kubernetes deployment targets in Octopus Deploy",
+		Example: heredoc.Docf("$ %s deployment-target Kubernetes create", constants.ExecutableName),
```
Comment (benPearce1) [inline pkg/cmd/target/kubernetes/kubernetes.go:18]: a controversial comment here: can we add an alias for `k8s`

Comment (domenicsim1) [reply pkg/cmd/target/kubernetes/kubernetes.go:18]: I like this idea

Comment (domenicsim1) [reply pkg/cmd/target/kubernetes/kubernetes.go]: Done.

**Comment by domenicsim1** [comment]
> there is a common `deployment-target view` command that attempts to defer to the specific implementations, this will need to be updated for to use the k8s view

Missed that, view updated.

File: pkg/cmd/target/kubernetes/create/create.go [hunk 2]
```diff
@@ -0,0 +1,1144 @@
...
+	flags.StringVar(&createFlags.AssumedRoleExternalID.Value, createFlags.AssumedRoleExternalID.Name, "", "AWS assumed role external ID.")
+	flags.StringVar(&createFlags.EKSClusterName.Value, createFlags.EKSClusterName.Name, "", "AWS EKS Cluster Name")
+
+	// Google Cloud Account
+	flags.BoolVar(&createFlags.UseVMServiceAccount.Value, createFlags.UseVMServiceAccount.Name, false, "When running in a Compute Engine virtual machine, use the associated VM service account.")
+	flags.BoolVar(&createFlags.ImpersonateServiceAccount.Value, createFlags.ImpersonateServiceAccount.Name, false, "Impersonate service accounts.")
+	flags.StringVar(&createFlags.ServiceAccountEmails.Value, createFlags.ServiceAccountEmails.Name, "", "Service Account Email.")
+	flags.StringVar(&createFlags.GKEClusterName.Value, createFlags.GKEClusterName.Name, "", "GKE Cluster Name.")
+	flags.StringVar(&createFlags.Project.Value, createFlags.Project.Name, "", "Project.")
+	flags.StringVar(&createFlags.ClusterType.Value, createFlags.ClusterType.Value, "", "Cluster Type.")
```
Comment (benPearce1) [inline pkg/cmd/target/kubernetes/create/create.go:288]:
```suggestion
	flags.StringVar(&createFlags.ClusterType.Value, createFlags.ClusterType.Name, "", "Cluster Type.")
```

File: pkg/cmd/target/kubernetes/create/create.go [hunk 3]
```diff
@@ -0,0 +1,1144 @@
...
+	flags.StringVar(&createFlags.Zone.Value, createFlags.Zone.Name, "", "Zone.")
+	flags.StringVar(&createFlags.Region.Value, createFlags.Region.Name, "", "Region.")
```
Comment (benPearce1) [inline pkg/cmd/target/kubernetes/create/create.go:290]: Can we make these a bit more descriptive

File: pkg/cmd/target/kubernetes/create/create.go [hunk 4]
```diff
@@ -0,0 +1,1144 @@
...
+	flags.IntVar(&createFlags.AssumedRoleSessionDuration.Value, createFlags.AssumedRoleSessionDuration.Name, 3600, "AWS assumed role session duration in seconds. (defaults to 3600 seconds, 1 hour)")
+	flags.StringVar(&createFlags.AssumedRoleExternalID.Value, createFlags.AssumedRoleExternalID.Name, "", "AWS assumed role external ID.")
+	flags.StringVar(&createFlags.EKSClusterName.Value, createFlags.EKSClusterName.Name, "", "AWS EKS Cluster Name")
+
+	// Google Cloud Account
+	flags.BoolVar(&createFlags.UseVMServiceAccount.Value, createFlags.UseVMServiceAccount.Name, false, "When running in a Compute Engine virtual machine, use the associated VM service account.")
+	flags.BoolVar(&createFlags.ImpersonateServiceAccount.Value, createFlags.ImpersonateServiceAccount.Name, false, "Impersonate service accounts.")
+	flags.StringVar(&createFlags.ServiceAccountEmails.Value, createFlags.ServiceAccountEmails.Name, "", "Service Account Email.")
+	flags.StringVar(&createFlags.GKEClusterName.Value, createFlags.GKEClusterName.Name, "", "GKE Cluster Name.")
+	flags.StringVar(&createFlags.Project.Value, createFlags.Project.Name, "", "Project.")
```
Comment (benPearce1) [inline pkg/cmd/target/kubernetes/create/create.go:287]: Can we make this a bit more descriptive

Comment (domenicsim1) [reply pkg/cmd/target/kubernetes/create/create.go]: The flag names will now help give this more meaning. The portal does not provide any more information about them, they are concepts for the targeted platform, i'm not sure how to make it more descriptive.

File: pkg/cmd/target/kubernetes/view/view.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+func ViewRun(opts *shared.ViewOptions) error {
+	return shared.ViewRun(opts, contributeEndpoint, "Kubernetes")
+}
+
+func contributeEndpoint(_ *shared.ViewOptions, targetEndpoint machines.IEndpoint) ([]*output.DataRow, error) {
+	data := []*output.DataRow{}
+	endpoint := targetEndpoint.(*machines.KubernetesEndpoint)
+
+	data = append(data, output.NewDataRow("Kubernetes Cluster URL", endpoint.ClusterURL.String()))
```
Comment (benPearce1) [inline pkg/cmd/target/kubernetes/view/view.go:45]: The cluster url property isn't used for all auth types
image


The view command should probably include the auth type, and perhaps some specific detail for each type

File: pkg/cmd/target/kubernetes/create/create_test.go [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+func getK8sStandaredDetailsPA() []*testutil.PA {
+	return []*testutil.PA{
+		testutil.NewInputPrompt("Kubernetes cluster URL", "Must be an absolute URL. e.g. https://kubernetes.example.com", "https://octopus.com"),
+		testutil.NewSelectPrompt("Select Certificate", "", testCertNames, testCertNames[1]),
+		testutil.NewConfirmPrompt("Skip TLS Verification", "Enable this option to skip the verification of the cluster certificate. This can only be selected if no cluster certificate is specified.", false),
+		testutil.NewInputPrompt("Kubernetes Namespace", "", "space"),
+	}
+}
+
+func assertK8sStandaredDetials(t *testing.T, flags *k8s.CreateFlags) {
```
Comment (benPearce1) [inline pkg/cmd/target/kubernetes/create/create_test.go:203]:
```suggestion
func assertK8sStandaredDetails(t *testing.T, flags *k8s.CreateFlags) {
```

File: pkg/cmd/target/kubernetes/create/create.go [hunk 7]
```diff
@@ -0,0 +1,1145 @@
...
+	FlagUseAdminCredentials  = "aks-use-admin-credentials"
+
+	// AWS Account
+	FlagUseServiceRole             = "eks-use-service-role"
+	FlagAssumeServiceRole          = "eks-assume-service-role"
+	FlagAssumedRoleARN             = "eks-assumed-role-arn"
+	FlagAssumedRoleSessionName     = "eks-assumed-role-session-name"
+	FlagAssumedRoleSessionDuration = "eks-assumed-role-session-duration"
+	FlagAssumedRoleExternalID      = "eks-assumed-role-external-id"
+	FlagEKSClusterName             = "eks-cluster-name"
```
Comment (benPearce1) [inline pkg/cmd/target/kubernetes/create/create.go:47]: These prefixes are way better, and didn't add much to the flags length 👍

**Review by benPearce1 — APPROVED** [review]
One comment on the view command, but other than that looks good

**Pull request merged**